src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -275,27 +275,34 @@ internal static class StreamHelper
 
         #region Static Methods
 
-        internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet, CancellationTokenSource cts)
+        internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet, CancellationToken cancellationToken)
         {
             if (cmdlet == null)
             {
                 throw new ArgumentNullException(nameof(cmdlet));
             }
 
-            Task copyTask = input.CopyToAsync(output, cts.Token);
+            Task copyTask = input.CopyToAsync(output, cancellationToken);
 
             ProgressRecord record = new ProgressRecord(
                 ActivityId,
                 WebCmdletStrings.WriteRequestProgressActivity,
                 StringUtil.Format(WebCmdletStrings.WriteRequestComplete, 0));
-            do
+            try
             {
-                record.StatusDescription = StringUtil.Format(WebCmdletStrings.WriteRequestComplete, output.Position);
-                cmdlet.WriteProgress(record);
+                do
+                {
+                    record.StatusDescription = StringUtil.Format(WebCmdletStrings.WriteRequestProgressStatus, output.Position);
+                    cmdlet.WriteProgress(record);
 
-                Task.Delay(1000).Wait(cts.Token);
+                    Task.Delay(1000).Wait(cancellationToken);
+                }
+                while (!copyTask.IsCompleted);
+            }
+            catch
```

Comment:
You should catch the `OperationCancellationException` only.

Reply:
Done.